test/Parse/matching_patterns.swift

```diff
@@ -289,16 +289,16 @@ case (let (_, _, _)) + 1:
   ()
 }
 
-// FIXME: We don't currently allow subpatterns for "isa" patterns that
-// require interesting conditional downcasts.
+// "isa" patterns.
 class Base { }
 class Derived : Base { }
 
 
 switch [Derived(), Derived(), Base()] {
-case let ds as [Derived]: // expected-error{{collection downcast in cast pattern is not implemented; use an explicit downcast to '[Derived]' instead}}
+case let ds as [Derived]:
```

Comment:
Could you please add a test-case that uses `switch` + `case let ... as ...` inside of a closure and SILGen test-case for that as well?

Reply:
Yes, I will address this

Reply:
@xedin I just pushed new changes. It has more tests!